src/compiler/checker.ts

```diff
@@ -15539,6 +15539,20 @@ namespace ts {
             return checkNonNullType(checkExpression(node), node);
         }
 
+        function checkNonNullCallLikeExpression(node: Expression | QualifiedName) {

```

Comment:
i do not think we should duplicate the function implementation here. consider passing the new error messages as optional arguments to checkNullExpression instead.

Reply:
@mhegazy thanks for the code review. I just sent a new commit that uses optional arguments instead of a new function.